### PR TITLE
Removed `discourse-gatekeeper/discourse-ahead-ok`

### PR DESCRIPTION
**Pull request by codethulu**
Applicable spec: N/A

### Overview

Removed `discourse-gatekeeper/discourse-ahead-ok` and related documentation, as this is no longer required.

### Rationale

Using `discourse-gatekeeper/discourse-ahead-ok` should no longer be possible.

### Module Changes

N/A

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] Changelog has been updated
- [x] Version has been incremented

### CODE REVIEW

**Review by jdkandersson — COMMENTED** [review]
Looks fine so far - I think some of the unit and integration tests still need to be fixed. Please also ensure that you do a repo search for the tag to make sure all references are removed

**Comment by codethulu** [comment]
ERROR    root:check.py:205 there is a problem preventing the execution of an action
path: doc
problem: detected unmerged community contributions, these need to be resolved before proceeding.  Paths with potentially unmerged community contributions: {'doc', 'alt-doc'}.

Getting this error at the moment which seems to be failing the integration test. Besides this, I believe that I have made the necessary changes. All unit tests are passing as expected.

**Comment by codethulu** [comment]
Tests just finished wrapping up. I believe that all necessary changes have been made, and this is ready for review.

File: tests/integration/test___init__run_conflict.py [hunk 1]
```diff
@@ -49,16 +49,14 @@ async def test_run_conflict(
...
-            changes with discourse-gatekeeper/discourse-ahead-ok applied
     assert: then:
         1. the documentation page is created
         2. the documentation page is not updated
         3. the documentation page is not updated
         4. the documentation page is updated
         5. the alternate documentation page is created
         6. the documentation page is not updated
-        6. the documentation page is updated
+        7. the documentation page is updated
```
Comment (cbartz) [inline tests/integration/test___init__run_conflict.py:59]: You have removed a test, is it still valid?

**Comment by Thanhphan1147** [comment]
Hi, is the deletion of files in `src-docs` intended?

**Review by cbartz — COMMENTED** [review]
Looks good in general!
But please run src-docs accordingly, it seems you have removed the contents of the src-docs folder.
Please also tick all the checkboxes (after applying/reading the content). 
Thank you very much.

**Review by arturo-seijas — DISMISSED** [review]

File: tests/integration/test___init__run_conflict.py (hunk 1 above)
Comment (codethulu) [reply tests/integration/test___init__run_conflict.py:59]: I have removed the test since it is now obsolete, due to the ahead tag being removed.

**Comment by codethulu** [comment]
Hmmm seems as though src-docs did not generate. I think this may be attributed due to some strange python path behaviour. I'll try fix it up and generate the src-docs now..

**Review by arturo-seijas — APPROVED** [review]

File: tests/integration/test___init__run_conflict.py (hunk 1 above)
Comment (cbartz) [reply tests/integration/test___init__run_conflict.py:59]: But I believe there are not 7 tests there anymore, so this line can be removed, right?

**Comment by codethulu** [comment]
Addressed conversation about comments in test file 👍

**Review by cbartz — APPROVED** [review]

**Review by javierdelapuente — APPROVED** [review]

**Pull request merged**